Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -155,6 +120,44 @@ - (void)trackRevenue:(NSDictionary *)properties andRevenueOrTotal:(NSNumber *)re
     }
 }
 
+- (void)trackLogRevenueV2:(NSDictionary *)properties andRevenueOrTotal:(NSNumber *)revenueOrTotal
+{
+    id price = [properties objectForKey:@"price"];
+    id quantity = [properties objectForKey:@"quantity"];
+
+    // if no price fallback to using revenue
+    if (!price || ![price isKindOfClass:[NSNumber class]]) {
+        price = revenueOrTotal;
+        quantity = [NSNumber numberWithInt:1];
+    } else if (!quantity || ![quantity isKindOfClass:[NSNumber class]]) {
+        quantity = [NSNumber numberWithInt:1];
+    }
+
+    [[self.amprevenue setPrice:price] setQuantity:[quantity integerValue]];
+    SEGLog(@"[[AMPRevenue revenue] setPrice:%@] setQuantity: %d];", price, [quantity integerValue]);
+
+    id productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"];
+    if (productId && [productId isKindOfClass:[NSString class]] && ![productId isEqualToString:@""]) {
+        [self.amprevenue setProductIdentifier:productId];
+        SEGLog(@"[[AMPRevenue revenue] setProductIdentifier:%@];", productId);
+    }
+
+    //Receipt is meant to be of type NSData
+    id receipt = [properties objectForKey:@"receipt"];
+    if (receipt && [receipt isKindOfClass:[NSString class]] && ![receipt isEqualToString:@""]) {
+        [self.amprevenue setReceipt:receipt];
+        SEGLog(@"[[AMPRevenue revenue] setReceipt:%@];", receipt);
+    }
+    id revenueType = [properties objectForKey:@"revenueType"] ?: [properties objectForKey:@"revenue_type"];
```

Comment:
Generally it's preferred to use shorthand syntax via `[]`. is there any reason you are not?

Reply:
It's in #40